superduperdb/ext/openai/model.py

```diff
@@ -71,18 +71,38 @@ def __post_init__(self):
         self.encoder = vector(self.shape)
 
     @retry
-    def _predict_one(self, X, **kwargs):
+    def predict_one(self, X: str, **kwargs):
         e = Embedding.create(input=X, model=self.identifier, **kwargs)
         return e['data'][0]['embedding']
 
+    @retry
+    async def apredict_one(self, X: str, **kwargs):
+        e = await Embedding.acreate(input=X, model=self.identifier, **kwargs)
+        return e['data'][0]['embedding']
+
     @retry
     def _predict_a_batch(self, texts, **kwargs):
         out = Embedding.create(input=texts, model=self.identifier, **kwargs)['data']
         return [r['embedding'] for r in out]
 
-    def _predict(self, X, batch_size=100, **kwargs):  # asyncio?
+    @retry
+    async def _apredict_a_batch(self, texts, **kwargs):
+        out = await Embedding.create(input=texts, model=self.identifier, **kwargs)[
+            'data'
+        ]
+        return [r['embedding'] for r in out]
+
+    def _predict(self, X, batch_size=100, **kwargs):
         if isinstance(X, str):
-            return self._predict_one(X)
+            return self.predict_one(X)
+        out = []
+        for i in tqdm.tqdm(range(0, len(X), batch_size)):
+            out.extend(self._predict_a_batch(X[i : i + batch_size], **kwargs))
+        return out
+
+    async def _apredict(self, X, batch_size=100, **kwargs):
+        if isinstance(X, str):
+            return await self.apredict_one(X)
         out = []
```

Comment:
I'm not entirely sure how to use the async version of `tqdm`. My quick fix is to just drop it for now, and we can add the async version in the future if there is a request for it.